components/omega/src/ocn/Eos.h

```diff
@@ -0,0 +1,290 @@
+   void computeSpecVol(Array2DReal SpecVol,
+                       const Array2DReal &ConservTemp,
+                       const Array2DReal &AbsSalinity,
+                       const Array2DReal &Pressure);
+
+   /// Compute displaced specific volume (for vertical displacement)
+   void computeSpecVolDisp(Array2DReal SpecVol,
+                       const Array2DReal &ConservTemp,
+                       const Array2DReal &AbsSalinity,
+                       const Array2DReal &Pressure,
+                       I4 KDisp, const std::string &DispType);
+
+   /// Initialize EOS from config and mesh
+   static I4 init();
+
+ private:
+   /// Private constructor
+    Eos(const std::string &Name, const HorzMesh *Mesh, int NVertLevels);
+   /// Private destructor
+    ~Eos();
+
+    static Eos* instance_; ///< Instance pointer
+
+    // Delete copy and move constructors and assignment operators
+    Eos(const Eos&) = delete;
+    Eos& operator=(const Eos&) = delete;
+    Eos(Eos&&) = delete;
+    Eos& operator=(Eos&&) = delete;
+
+   I4 NCellsAll;         ///< Number of horizontal cells
+   I4 NChunks;           ///< Number of vertical chunks (for vectorization)
+   const int NVertLevels;///< Number of vertical levels
+
+   /// Compute linear EOS specific volume for a chunk
+   KOKKOS_FUNCTION void computeSpecVolLinear(Array2DReal SpecVol,
+                                             I4 ICell, I4 KChunk,
+                                             const Array2DReal &ConservTemp,
+                                             const Array2DReal &AbsSalinity) {
+      const I4 KStart = KChunk * VecLength;
+      for (int KVec = 0; KVec < VecLength; ++KVec) {
+         const I4 K = KStart + KVec;
+         SpecVol(ICell, K) =
+             1.0 / (RhoT0S0 + (DRhodT * ConservTemp(ICell, K) +
+                               DRhodS * AbsSalinity(ICell, K)));
+      }
+   }
+
+   /// Compute TEOS-10 specific volume for a chunk
+   KOKKOS_FUNCTION void computeSpecVolTeos10(Array2DReal SpecVol,
+                                             I4 ICell, I4 KChunk,
+                                             const Array2DReal &ConservTemp,
+                                             const Array2DReal &AbsSalinity,
+                                             const Array2DReal &Pressure,
+                                             I4 KDisp, const std::string &DispType) {
+      const I4 KStart = KChunk * VecLength;
+      for (int KVec = 0; KVec < VecLength; ++KVec) {
+         const I4 K = KStart + KVec;
+
+         calcPCoeffs(SpecVolPCoeffs, KVec, ConservTemp(ICell, K), AbsSalinity(ICell, K));
+
+         if (KDisp == 0) {
+            // No displacement
+            SpecVol(ICell, K) = calcRefProfile(Pressure(ICell, K)) +
+                              calcDelta(SpecVolPCoeffs, KVec, Pressure(ICell, K));
+         } else {
+            // Displacement, use the displaced pressure
+            if (DispType == "absolute") {
```

Comment:
As far as I am aware, we do not have a case where we need the absolute displacement type. My understanding was that in MPAS-O, this was used to displace to the surface, which we want to treat differently in Omega (i.e. not parcel up to the surface, but the integration from the surface down). Can anyone think of a case where we want the absolute? 
I don't mind leaving it in the code for now, I simply don't want it to be used if we don't think we should use it. 
My understanding of the use cases are: 1) local displacement for stability calculation, 2) displacement relative to the surface for the vertical mixing parameterizations, which we had agreed can be done by applying `computeSpecVol` to the pre-processed integrated T,S fields. 
Did I forget something obvious?

Reply:
You're right, @alicebarthel, I misplaced my notes and forgot about what we had agreed upon. I do not think we need the absolute displacement. I will take that option away.

On that note, do we want to have two functions, `computeSpecVol` and `computeSpecVolDisp`? In MPAS we only had one and let the value of KDisp dictate whether it was the displaced version or not.